multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/listeners/EnterTestingPhaseListenerTest.java

```diff
@@ -0,0 +1,100 @@
+package org.cloudfoundry.multiapps.controller.process.listeners;
+
+import org.cloudfoundry.multiapps.controller.api.model.ImmutableOperation;
+import org.cloudfoundry.multiapps.controller.api.model.Operation;
+import org.cloudfoundry.multiapps.controller.core.util.ApplicationConfiguration;
+import org.cloudfoundry.multiapps.controller.persistence.query.impl.OperationQueryImpl;
+import org.cloudfoundry.multiapps.controller.persistence.services.HistoricOperationEventService;
+import org.cloudfoundry.multiapps.controller.persistence.services.OperationService;
+import org.cloudfoundry.multiapps.controller.persistence.services.ProcessLoggerProvider;
+import org.cloudfoundry.multiapps.controller.persistence.services.ProcessLogsPersister;
+import org.cloudfoundry.multiapps.controller.persistence.services.ProgressMessageService;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.flowable.FlowableFacade;
+import org.cloudfoundry.multiapps.controller.process.util.MockDelegateExecution;
+import org.cloudfoundry.multiapps.controller.process.util.StepLogger;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+class EnterTestingPhaseListenerTest {
+
+    // needed because of @InjectMocks
```

Comment:
This could be avoided if you use constructor injection and create new instance via constructor, then probably you won't need to mock all object fields.

Reply:
As we discussed in the chat, it is used field which should be initialized, so avoiding field injection is not possible in this moment.